### PR TITLE
Remove `_` from task role name, which fix #47 of openpai-runtime.

### PR DESCRIPTION
**Pull request by siaimes**
1. remove `_` from task role name; 2. update regx to match `/^[a-z][a-z0-9]*$/`.

fix https://github.com/microsoft/openpai-runtime/issues/47

Signed-off-by: siaimes <34199488+siaimes@users.noreply.github.com>

### CODE REVIEW

File: src/webportal/src/app/job-submission/components/form-text-field.jsx [hunk 1]
```diff
@@ -30,8 +30,6 @@ import { BasicSection } from './basic-section';
 import { FormShortSection } from './form-page';
 import { DebouncedTextField } from './controls/debounced-text-field';
 
-const TEXT_FILED_REGX = /^[A-Za-z0-9\-._~]+$/;
```
Comment (Binyang2014) [inline src/webportal/src/app/job-submission/components/form-text-field.jsx:33]: Why remove this line?

Comment (siaimes) [reply src/webportal/src/app/job-submission/components/form-text-field.jsx:33]: Username and taskrole need to use different regulars, so use the `regx` parameter to pass it in.

File: src/webportal/src/app/job-submission/components/job-information.jsx [hunk 2]
```diff
@@ -33,6 +33,8 @@ import Card from '../../components/card';
 import { JobBasicInfo } from '../models/job-basic-info';
 import { PROTOCOL_TOOLTIPS } from '../utils/constants';
 
+const JOB_NAME_REGX = /^[A-Za-z0-9\-_]+$/;
```
Comment (Binyang2014) [inline src/webportal/src/app/job-submission/components/job-information.jsx:36]: Please use `/^[A-Za-z0-9\-._~]+$/` for job name

File: src/webportal/src/app/job-submission/components/tab-form-content.jsx [hunk 3]
```diff
@@ -39,6 +39,8 @@ import { getDefaultContainerSize } from '../models/container-size';
 import { PROTOCOL_TOOLTIPS } from '../utils/constants';
 import config from '../../config/webportal.config';
 
+const TASK_ROLE_NAME_REGX = /^[a-z][a-z0-9]*$/;
```
Comment (Binyang2014) [inline src/webportal/src/app/job-submission/components/tab-form-content.jsx:42]: Please refer https://github.com/microsoft/frameworkcontroller/blob/d602cdcbab2b91a7772f9c0d956d17e5f9d69847/pkg/apis/frameworkcontroller/v1/crd.go#L33 for task name

File: src/webportal/src/app/job-submission/models/protocol-schema.js [hunk 4]
```diff
@@ -67,7 +67,7 @@ export const taskRoleSchema = Joi.object().keys({
 });
 
 export const taskRolesSchema = Joi.object().pattern(
-  /^[a-zA-Z_][a-zA-Z0-9_]*$/,
+  /^[a-z][a-z0-9]*$/,
```
Comment (Binyang2014) [inline src/webportal/src/app/job-submission/models/protocol-schema.js:70]: Please refer https://github.com/microsoft/frameworkcontroller/blob/d602cdcbab2b91a7772f9c0d956d17e5f9d69847/pkg/apis/frameworkcontroller/v1/crd.go#L33 for task role name

**Review by Binyang2014 — APPROVED** [review]

**Comment by Binyang2014** [comment]
@siaimes Thanks for this fix

**Pull request merged**